src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs

```diff
@@ -65,11 +58,12 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
+            ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ElasticsearchActivitySourceHelper.ActivitySource);
+            ActivityInstrumentationHelper.SetKindProperty(activity, ActivityKind.Client);
+
             var method = this.methodFetcher.Fetch(payload);
             activity.DisplayName = this.GetDisplayName(activity, method);
 
-            this.activitySource.Start(activity, ActivityKind.Client);
```

Comment:
If the intention is to change the `ActivitySource` of this activity to `ElasticsearchActivitySourceHelper.ActivitySourceName` and update the `ActivityKind` to `Client` then you should add these two lines here:
```
ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ElasticsearchActivitySourceHelper.ActivitySourceName);
ActivityInstrumentationHelper.SetKindProperty(activity, ActivityKind.Client);
```

Reply:
its there. lines 61,62 :)

Reply:
Oops! Never mind!